src/utils/web.vala

```diff
@@ -51,9 +51,12 @@ namespace ProtonPlus.Utils {
 
                 FileOutputStream output_stream = yield file.create_async (FileCreateFlags.REPLACE_DESTINATION, Priority.DEFAULT, null);
 
-                // Temporary fix for GitLab since they don't give the file size from their API
-                if (download_size == -1) {
-                    download_size = soup_message.get_response_headers ().get_content_length ();
+                // Prefer real Content-Length header from the server if it exists.
+                // NOTE: Servers typically return "0" when it doesn't know (for
+                // live-generated files, such as GitHub's commit-based source archives).
+                var server_download_size = soup_message.get_response_headers ().get_content_length ();
+                if (server_download_size > 0) {
+                    download_size = server_download_size;
```

Comment:
We need to keep it like it was because most of the time I pass the proper download size so there's no need for that in most cases.
The few cases where I don't now the file size before hand I passed -1 to maybe have luck and be able to get the size with the content_length.
Doing it like you`re doing right now will not work in most cases as far as I remember with most direct download links I tested.
I may be wrong though so I will test a few links tonight to see if we can keep it the way you modified it.

Reply:
The server is the authority about the download size. It's always better than the "assumed size". That is why, if the server tells us the filesize, we use the server's answer rather than the assumption. This change is better.

PS: GitHub does not tell us the size of SteamTinkerLaunch archives. It is always `Content-Length: 0`, due to it streaming the creation of the zip in realtime.

Reply:
I have also confirmed that GitHub sends `Content-Length` when you request actual "assets" from proper releases.

Example:

- https://github.com/GloriousEggroll/proton-ge-custom/releases/tag/GE-Proton9-13
- File: https://github.com/GloriousEggroll/proton-ge-custom/releases/download/GE-Proton9-13/GE-Proton9-13.tar.gz
- Server header: `Content-Length: 430084551`

Reply:
I'll remove the code to get the size before hand then
I have no problem with that modification